test.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+set -e
+clean_up () {
+    ARG=$?
+
+    if [[ $ARG -ne 0 ]]; then
+      echo "Test failed, showing docker logs"
+      echo "- Mockserver"
+      docker-compose -f ./test/docker-compose.yml logs mockserver
+      echo "- Fluent Bit"
+      docker-compose -f ./test/docker-compose.yml logs newrelic-fluent-bit-output
+    fi
+
+    echo "Cleaning up"
+    rm -r ./test/testdata || true
+    docker-compose -f ./test/docker-compose.yml down
+
+    exit $ARG
+}
+trap clean_up EXIT
+
+function check_logs {
+  curl -X PUT -s --fail "http://localhost:1080/mockserver/verify" -d @test/verification.json >> /dev/null
+  RESULT=$?
+  return $RESULT
+}
+
+function check_mockserver {
+  curl -X PUT -s --fail "http://localhost:1080/mockserver/status" >> /dev/null
+  RESULT=$?
+  return $RESULT
+}
+
+# Create testdata folder and log file
+mkdir ./test/testdata || true
+touch ./test/testdata/fbtest.log
+
+# Initialize
+if [ ${CI:-no} = "no" ]; then
+  echo "Building docker image"
+  docker build -f ${DOCKERFILE:-Dockerfile} -t fb-output-plugin .
+fi
+
+echo "Starting docker compose"
+docker-compose -f ./test/docker-compose.yml up -d
+
+# Waiting mockserver to be ready
+max_retry=10
+counter=0
+until check_mockserver
+do
+  echo "Waiting mockserver to be ready. Trying again in 2s. Try #$counter"
+  sleep 2
+  [[ $counter -eq $max_retry ]] && echo "Mockserver failed to start!" && exit 1
+  counter+=1
+done
```

Comment:
[❓] Out of ignorance: could we be using [this](https://docs.docker.com/compose/compose-file/compose-file-v2/#healthcheck)?

Reply:
Then the thing would be if docker-compose started in daemon mode can wait for all the containers to be healthy before returning, which I don' tknow. 🤔

Reply:
Hi @jsubirat, yes, in fact that was my first approach, the problem is that mockserver doesn't come with curl/wget or other utilities to perform the healthcheck because of that we need to do perform the healthcheck from outside in a more "manually" way.

There is a ticket on mockserver to allow that https://trello.com/c/qYxmdShC/147-add-support-for-docker-healthcheck but... not so much movement there.